### DISCUSSION TITLE
User Retention with Vercel Analytics

### DISCUSSION THREAD

**Discussion by ramikalai** [vercel/community#4814] [Feature Question] [answered]
### Question

Hi,

Is there a way to get an idea of user retention using Vercel analytics? How do I know if the visits today are new or returning users?

Maybe there's some way of doing this with custom events?

I want to use Vercel Analytics but it feels so sparse in terms of features. Visit duration is another one that would be nice to have.

**Comment by kidow** [comment]
Half a year has passed and there are still no comments. Vercel is a really good tool, but Analytics still needs more features. I hope the user retention feature is added as soon as possible.

**Comment by timolins** [comment, marked as answer]
Hi! We appreciate your suggestion. Due to the privacy-friendly nature of how Vercel Analytics tracks visitors, it is not possible to track users across multiple days.

However, we have been experimenting with opt-in cookies, which can unlock features like retention in the future to offer the best of both worlds.

We have plans to add engagement metrics like visit duration and bounce rate, but we don't have a precise ETA for these updates yet.
